delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/ResourceF.scala

```diff
@@ -54,7 +58,7 @@ final case class ResourceF[A](
   // format: off
   override def equals(obj: Any): Boolean =
     obj match {
-      case b @ ResourceF(_, _, `rev`, `types`, `deprecated`, `createdAt`, `createdBy`, `updatedAt`, `updatedBy`, `schema`, `value`) =>
+      case b @ ResourceF(_, _, `rev`, `types`, `deprecated`, `createdAt`, `createdBy`, `updatedAt`, `updatedBy`, `schema`, `value`, _, _) =>
```

Comment:
You can, and probably should, add the incoming and outgoing to the computation of the equals and hashcode.
The way to do it is pretty similar to the `id` and `accessUrl`

Reply:
I talked about it before but why not having something like this in ResourceF instead of the function:
```
sealed trait ResourceLink
final case class FixedLink(uri: Uri)
final case class CompactableLink(uri: Uri, mappings: ApiMappings, base: ProjectBase)
//describing other type of links
```
We would get rid of the functions where some have default values and therefore we wouldn't need to override the equals and hashCode.
And we would compute the different links in the encoder of ResourceLink and merge the resulting json with the rest of the resource fields

Reply:
I'm not sure if exactly that would work, since you need the `baseUri` to build those `uri` that you mention. The `ResourceLink` has to be built in order to build a `ResourceF`...so every time you build a resourceF you'll end up with the need of an `implicit baseUri: BaseUri`.

I don't see any simple way to do it without having the already mentioned drawback

Reply:
It would be paths more than uris.
And we would build the different links when encoding a little like what we do with subjects for the createdBy/updatedBy.

Reply:
With paths works. But now you have the logic scattered between `ResourceLink` and `AccessUrl`

Reply:
Maybe something directly in `AccessUrl` or next to it called `PartialAccessUrl`, which interact with one another could work

Reply:
The scattered logic should be something we can solve